### ISSUE TITLE
Adding WPA2 Entreprise

### ISSUE THREAD

**Issue by irgusite** [rockymeza/wifi#52]
Would it be possible to add WPA2 Entreprise support?

**Comment by MrDadaGuy** [comment]
+1 I would also like to see this implemented - thanks!

**Comment by DamianPe** [comment]
Hi! Is there any progres regarding Enterprise, please. This would make your work even more amazing!
Cheers!

**Comment by irgusite** [comment]
Actually not on my side, I had problems with my WPA2 entreprise installation and have not worked on it. I have only integrated the detection of the WPA2/entreprise. 

Le 8 janvier 2016 20:32:26 GMT+01:00, DamianPe <notifications@github.com> a écrit :
>Hi! Is there any progres regarding Enterprise, please. This would make
>your work even more amazing!
>Cheers!
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/rockymeza/wifi/issues/52#issuecomment-170100896

-- 
Envoyé de mon appareil Android avec K-9 Mail. Veuillez excuser ma brièveté.

**Comment by MrDadaGuy** [comment]
Hi - I was watching this on github and saw your note.  I ended up rolling
my own Enterprise mode for a different project.  I didn't break it out to a
separate module but it should be easy enough to see & copy what I've done:

https://bitbucket.org/veggiebenz/pyspresso/src/1a15df69102ae96abd5fd352267445bd9dab0bae/ispresso.py?fileviewer=file-view-default


Starting around line 688 thru 833


Good luck, hope this helps.


Chris

**Comment by DamianPe** [comment]
Hello irgusite,

Thanks for fast response and sorry for mine late one.
Is detection of WPA2 Enterprise included in main branch of your code? I have one available and when I read "encryption_type" it returns only "wpa2", exactly the same as while asking about regular WPA2 network (my personal one)...

Hi MrDadaGuy,

Thanks for your code, I appreciate your help, I didn't expect such a response :)
When I deal with detection of WPA2 Enterprise, I will take your code as an example :)

**Comment by irgusite** [comment]
Hi,
Yes, my main branch includes the detection. I don't know if I've pushed the code on github. I look at it when I'm back home :)

Le 17 janvier 2016 12:19:12 GMT+01:00, DamianPe <notifications@github.com> a écrit :
>Hello irgusite,
>
>Thanks for fast response and sorry for mine late one.
>Is detection of WPA2 Enterprise included in main branch of your code? I
>have one available and when I read "encryption_type" it returns only
>"wpa2", exactly the same as while asking about regular WPA2 network (my
>personal one)...
>
>Hi MrDadaGuy,
>
>Thanks for your code, I appreciate your help, I didn't expect such a
>response :)
>When I deal with detection of WPA2 Enterprise, I will take your code as
>an example :)
>
>
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/rockymeza/wifi/issues/52#issuecomment-172314964

-- 
Envoyé de mon appareil Android avec K-9 Mail. Veuillez excuser ma brièveté.

**Comment by irgusite** [comment]
@DamianPe So, yeah I've checked and the detection is included. The last thing to do is correct the integration tests. Because i've no WPA2/Entreprise wifi I can't correct it. Look at 88d62fdc5fde36472ff0d011a6c5c11e23dc3a6d and 5e3805f98e29c57a7fd8ce240c444aa163c834d4 to see the modifications ;)

**Comment by DamianPe** [comment]
Thanks for checking that for me.
I tried to install tkat specific commit (the one with spelling errors
corrected) but for some reason it keeps returning an error. I have no idea
where is the fault. Can you quide me through, please?

2016-01-17 17:07 GMT+01:00 irgusite <notifications@github.com>:

**Comment by irgusite** [comment]
@DamianPe I can. But waht error is it returning?

**Comment by DamianPe** [comment]
@irgusite 

Copy-paste from PuTTy:
root@raspberry:/home/pi#  pip install git+git://github.com/rockymeza/wifi.git@5e                                                     3805f98e29c57a7fd8ce240c444aa163c834d4
Downloading/unpacking git+git://github.com/rockymeza/wifi.git@5e3805f98e29c57a7f                                                     d8ce240c444aa163c834d4
  Cloning git://github.com/rockymeza/wifi.git (to 5e3805f98e29c57a7fd8ce240c444a                                                     a163c834d4) to /tmp/pip-MErJ9y-build
  Could not find a tag or branch '5e3805f98e29c57a7fd8ce240c444aa163c834d4', ass                                                     uming commit.
fatal: reference is not a tree: 5e3805f98e29c57a7fd8ce240c444aa163c834d4
  Complete output from command /usr/bin/git checkout -q 5e3805f98e29c57a7fd8ce24                                                     0c444aa163c834d4:

----------------------------------------
Command /usr/bin/git checkout -q 5e3805f98e29c57a7fd8ce240c444aa163c834d4 failed                                                      with error code 128 in /tmp/pip-MErJ9y-build
Storing complete log in /root/.pip/pip.log

**Comment by irgusite** [comment]
The commit doesn't exists on this repo as it's not merged. Try maybe git@github.com:irgusite/wifi.git.

**Comment by DamianPe** [comment]
@irgusite 
This worked perfectly, thank you! I Should have asked you several hours earlier ;)

**Comment by DamianPe** [comment]
@irgusite 
I have a problem again, unfortunately. Everything worked beautifully last time but my Raspberry Pi started to crash at boot and I had to format SD card, install brand new OS and I'm now restoring my config. I'm stuck at the commit of yours, I keep getting different errors from github, last one is: 
```
ERROR: Repository not found.
fatal: Could not read from remote repository.
```
What am I doing wrong this time? Or maybe you have changed something?

**Comment by irgusite** [comment]
What command are you using? I've changed nothing since the last time :)

Le 1 février 2016 11:55:21 GMT+01:00, DamianPe <notifications@github.com> a écrit :
>@irgusite 
>I have a problem again, unfortunately. Everything worked beautifully
>last time but my Raspberry Pi started to crash at boot and I had to
>format SD card, install brand new OS and I'm now restoring my config.
>I'm stuck at the commit of yours, I keep getting different errors from
>github, last one is: 
>```
>ERROR: Repository not found.
>fatal: Could not read from remote repository.
>```
>What am I doing wrong this time? Or maybe you have changed something?
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/rockymeza/wifi/issues/52#issuecomment-177913555

-- 
Envoyé de mon appareil Android avec K-9 Mail. Veuillez excuser ma brièveté.

**Comment by DamianPe** [comment]
I am using
` pip install git+ssh://git@github.com:irgusite/wifi.git@5e3805f98e29c57a7fd8ce240c444aa163c834d4`

**Comment by irgusite** [comment]
Okay so, I've tried and I can't clone the repo with ssh but it works with https so try with that and telle me if it's working. Could you also supply the iwlist scan output? Would be nice so I can add the test and and we could merge the changes to the main branch ;)

**Comment by DamianPe** [comment]
OK, it worked with https, thanks!
Would you like to receive output from scan here or maybe on private? I don't think it should be pasted here :)

**Comment by irgusite** [comment]
Yes, send me a pm. Thank you :)

Le 1 février 2016 13:21:02 GMT+01:00, DamianPe <notifications@github.com> a écrit :
>OK, it worked with https, thanks!
>Would you like to receive output from scan here or maybe on private? I
>don't think it should be pasted here :)
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/rockymeza/wifi/issues/52#issuecomment-177947397

-- 
Envoyé de mon appareil Android avec K-9 Mail. Veuillez excuser ma brièveté.

**Comment by DamianPe** [comment]
PMs on Github? Rather impossible. Maybe an e-mail but I don't have yours...
yet :)

2016-02-01 13:22 GMT+01:00 irgusite <notifications@github.com>:

**Comment by irgusite** [comment]
irgusite (at) gmail ;)

**Comment by DamianPe** [comment]
@irgusite 
Have you received my e-mail?

**Comment by irgusite** [comment]
Yes I have, it seems you haven't got my answer. :) I will send it back :)

Le 3 février 2016 08:38:10 GMT+01:00, DamianPe <notifications@github.com> a écrit :
>@irgusite 
>Have you received my e-mail?
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/rockymeza/wifi/issues/52#issuecomment-179067672

-- 
Envoyé de mon appareil Android avec K-9 Mail. Veuillez excuser ma brièveté.

**Comment by DamianPe** [comment]
Nope, I checked SPAM folder, nothing there from you either... Weird...
